llama_stack/providers/remote/vector_io/weaviate/weaviate.py

```diff
@@ -247,7 +358,7 @@ async def register_vector_db(
                 ],
             )
 
-        self.cache[sanitized_collection_name] = VectorDBWithIndex(
+        self.cache[vector_db.identifier] = VectorDBWithIndex(
```

Comment:
Looks like this was a bug?

Reply:
Yes, I believe this was a bug. Since we just added Weaviate to the integration tests in this PR, this wasn't caught until now.